Lib/Src/CController.cpp

```diff
@@ -78,10 +79,11 @@ void CController::run()
 /**
  * @brief Give a new command to controller.
  *
- * @param p_command Pointer to NULL terminated command string.
+ * @param p_command Pointer to a command object that contains command name and
+ * arguments.
  * @return True if command was recognised.
  */
-bool CController::newCommand(etl::string<MAX_STRING_SIZE> command, IComChannel *p_comchannel)
+bool CController::newCommand(ICommand *p_command, IComChannel *p_comchannel)
```

Comment:
`bool CController::newCommand(ICommand::command_t command)`
That's the way I see this working, then the controller just compares `command.name` to a set table of commands.